src/datadog/datadog_agent.cpp

```diff
@@ -156,10 +157,36 @@ DatadogAgent::DatadogAgent(
       flush_interval_(config.flush_interval),
       request_timeout_(config.request_timeout),
       shutdown_timeout_(config.shutdown_timeout),
-      remote_config_(tracer_signature, rc_listeners, logger),
-      tracer_signature_(tracer_signature) {
+      remote_config_(tracer_signature, rc_listeners, logger) {
   assert(logger_);
 
+  // Set HTTP headers
+  headers_.emplace("Content-Type", "application/msgpack");
+  headers_.emplace("Datadog-Meta-Lang", "cpp");
+  headers_.emplace("Datadog-Meta-Lang-Version",
+                   tracer_signature.library_language_version);
+  headers_.emplace("Datadog-Meta-Tracer-Version",
+                   tracer_signature.library_version);
+
+  // Origin Detection headers are not necessary when Unix Domain Socket (UDS)
+  // is used to communicate with the Datadog Agent.
+  if (!contains(config.url.scheme, "unix")) {
+    if (auto container_id = container::get_id()) {
+      if (container_id->type == container::ContainerID::Type::container_id) {
+        headers_.emplace("Datadog-Container-ID", container_id->value);
+        headers_.emplace("Datadog-Entity-Id", "ci-" + container_id->value);
```

Comment:
```suggestion
        headers_.emplace("Datadog-Entity-Id", "cid-" + container_id->value);
```

According to [JIRA spec](https://datadoghq.atlassian.net/wiki/spaces/CONTP/pages/3731783774/New+Origin+Detection+Spec+for+Traces+and+DogStatsD)

Reply:
TBH, the spec is confusing. It mentions `Datadog-Entity-Id=ci-<CONTAINER_ID>` but for **backward compatibility** Agent would still need to be able to resolve `Datadog-Entity-Id=cid-<CONTAINER_ID>`, then proceed to talk about backward compatibility depending on `DD_EXTERNAL_ENV` existence _BUT_ that section is in red which means according to the color coding:
> Red solutions are solutions that have been refused

All in all, my understanding is th expected value is `Datadog-Entity-Id=ci-<CONTAINER_ID>` but for backward compatibility reason with the Agent(?) it must also/or support `Datadog-Entity-Id=cid-<CONTAINER_ID>`. @wdhif can you weight in?

Reply:
Thanks @wdhif